spec/services/edge_cache/bust_spec.rb

```diff
@@ -30,22 +27,12 @@
         stub_nginx
       end
 
-      let(:cache_bust_service) { described_class.new(path) }
-
       it "can bust a fastly cache" do
         allow(fastly_provider_class).to receive(:call)
 
-        cache_bust_service.call
-        expect(cache_bust_service.provider).to eq("fastly")
+        described_class.call(path)
```

Comment:
Can we test the expected `determine_provider_class` here as well? Just thinking that if we remove [this line](https://github.com/forem/forem/pull/11684/files#diff-bf52b55fce7912a9ec4a31f89a3f3e0face4025c68f5ce26da00c75c963163f8L39) we don't have any test that asserts that the provider we expect to be returned by the class is actually correct (but now that we've made it `protected`, I think we'd have to use `public_send` here)

Reply:
On second thought, I'm thinking maybe we revisit this in another PR once the refactor is more fleshed out because 1) `determine_provider_class` is now a `private` method and 2) it's tested in a roundabout way [here](https://github.com/forem/forem/pull/11684/files#diff-bf52b55fce7912a9ec4a31f89a3f3e0face4025c68f5ce26da00c75c963163f8R34). If the `determine_provider_class` didn't work as expected, this spec would fail because it wouldn't end up calling the correct class.

How does that sound @vaidehijoshi?

Reply:
That's fair! I can get down with punting this to another PR with more refactoring done :)